### PR TITLE
Convert DeviceInfoModuleTest to Kotlin

### PR DESCRIPTION
**Pull request by okwasniewski**
## Summary:

This PR converts DeviceInfoModuleTest into Kotlin as requested in https://github.com/facebook/react-native/issues/37708.

## Changelog:

[INTERNAL] [CHANGED] - Convert DeviceInfoModuleTest to Kotlin

## Test Plan:

1. Run `./gradlew :packages:react-native:ReactAndroid:test`.
2. All tests should pass.

### CODE REVIEW

**Review by cortinico — APPROVED** [review]
LGTM 👍 Thanks for sending this over @okwasniewski

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/deviceinfo/DeviceInfoModuleTest.kt [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  @After
+  fun teardown() {
+    DisplayMetricsHolder.setWindowDisplayMetrics(null)
+    DisplayMetricsHolder.setScreenDisplayMetrics(null)
+  }
+
+  @Test
+  fun test_itDoesNotEmitAnEvent_whenDisplayMetricsNotChanged() {
+    givenDisplayMetricsHolderContains(fakePortraitDisplayMetrics)
+    deviceInfoModule.typedExportedConstants
```
Comment (rshest) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/deviceinfo/DeviceInfoModuleTest.kt:64]: Is this supposed to have side effects?.. Looks confusing.

Comment (rshest) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/deviceinfo/DeviceInfoModuleTest.kt:64]: Guess it was just there to check that the getter doesn't crash, or something...

Comment (cortinico) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/deviceinfo/DeviceInfoModuleTest.kt:64]: Yup that's the reason behind this

**Review by cortinico — APPROVED** [review]

**Pull request closed without merging**